pepper-apis/auth0-config/rules/register-user-in-pepper.js

```diff
@@ -103,6 +103,12 @@ function (user, context, callback) {
 
         console.log(context);
 
+        // This is the token renewal case. Let's avoid going through pepper registration
+        if (context.request.query.renew_token_only) {
+            context.idToken[pepperUserGuidClaim] = user.app_metadata.user_guid;
+            return callback(null, user, context);
+        }
+
```

Comment:
Just 3 lines of code! But part of what was happening was that ALL token renewal requests ("Click Continue to extend your session") where generating calls to the pepper '/register route' . In the bug situation, the request contained a auth0 user id with the study id for the wrong study (the study in the other browser tab).

With this code, a token renewal never registers anyone. Note that we are counting on a new query parameter to identify the type of request.